xm-commons-domainevent-db/src/main/java/com/icthh/xm/commons/domainevent/db/service/DatabaseSource.java

```diff
@@ -0,0 +1,227 @@
+
+            JpaEntityContext context = JpaEntityContext.builder()
+                .entity(entity)
+                .id(id)
+                .currentState(currentState)
+                .previousState(previousState)
+                .propertyNames(propertyNames)
+                .types(types)
+                .domainEventOperation(UPDATE)
+                .build();
+
+            if (isIntercepted(tableName, sourceConfig, context)) {
+                DomainEvent dbDomainEvent = jpaEntityMapper.map(context);
+                eventPublisher.publish(DB.name(), dbDomainEvent);
+            }
+        }
+
+        return super.onFlushDirty(entity, id, currentState, previousState, propertyNames, types);
+    }
+
+    @Override
+    public boolean onSave(Object entity, Serializable id, Object[] state, String[] propertyNames, Type[] types) {
+        SourceConfig sourceConfig = xmDomainEventConfiguration.getSourceConfig(DB.name());
+
+        if (sourceConfig.isEnabled()) {
+
+            String tableName = findTableName(entity);
+            log.trace("onSave: tableName: {}, id: {}", tableName, id);
+
+            JpaEntityContext context = JpaEntityContext.builder()
+                .entity(entity)
+                .id(id)
+                .currentState(state)
+                .previousState(null)
+                .propertyNames(propertyNames)
+                .types(types)
+                .domainEventOperation(CREATE)
+                .build();
+
+            if (isIntercepted(tableName, sourceConfig, context)) {
+                DomainEvent dbDomainEvent = jpaEntityMapper.map(context);
+                eventPublisher.publish(DB.name(), dbDomainEvent);
+            }
+        }
+
+        return super.onSave(entity, id, state, propertyNames, types);
+    }
+
+    @Override
+    public void onDelete(Object entity, Serializable id, Object[] state, String[] propertyNames, Type[] types) {
+        SourceConfig sourceConfig = xmDomainEventConfiguration.getSourceConfig(DB.name());
+
+        if (sourceConfig.isEnabled()) {
+
+            String tableName = findTableName(entity);
+            log.trace("onDelete: tableName: {}, id: {}", tableName, id);
+
+            JpaEntityContext context = JpaEntityContext.builder()
+                .entity(entity)
+                .id(id)
+                .currentState(null)
+                .previousState(state)
+                .propertyNames(propertyNames)
+                .types(types)
+                .domainEventOperation(DELETE)
+                .build();
+
+            if (isIntercepted(tableName, sourceConfig, context)) {
+                DomainEvent dbDomainEvent = jpaEntityMapper.map(context);
+                eventPublisher.publish(DB.name(), dbDomainEvent);
+            }
+        }
+
+        super.onDelete(entity, id, state, propertyNames, types);
+    }
+
+    private String findTableName(Object entity) {
+        Table tableAnnotation = entity.getClass().getAnnotation(Table.class);
```

Comment:
I propose to cache table names. Number on entities is limited,  so cashe  entity.getClass().getSimpleName() -> tableName will be very useful